pkg/lib/userexport/service.go

```diff
@@ -199,8 +212,58 @@ func (s *UserExportService) ExportRecords(ctx context.Context, request *Request)
 		}
 	}
 
-	// TODO: Upload tmp result output to cloud storage
+	key := url.QueryEscape(fmt.Sprintf("%s-%s-%s.%s", task.AppID, task.ID, s.Clock.NowUTC().Format("20060102150405Z"), "ndjson"))
+	_, err = s.UploadResult(key, tmpResult)
+
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func (s *UserExportService) UploadResult(key string, resultFile *os.File) (response *http.Response, err error) {
+	file, err := os.Open(resultFile.Name())
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return nil, err
+	}
+
+	presignUploadRequest := cloudstorage.PresignUploadRequest{
+		Key:     key,
+		Headers: map[string]interface{}{},
+	}
+	presignUploadRequest.Headers["content-length"] = strconv.FormatInt(fileInfo.Size(), 10)
+	presignUploadRequest.Headers["content-type"] = "application/x-ndjson"
+	presignUploadRequest.Headers["Content-Disposition"] = fmt.Sprintf("attachment; filename=%s", key)
+	presignUploadResponse, err := s.CloudStorage.PresignPutRequest(&presignUploadRequest)
+	if err != nil {
+		return
+	}
+
+	buffer := bytes.NewBuffer(nil)
+	if _, err := io.Copy(buffer, file); err != nil {
+		return nil, err
+	}
```

Comment:
Do not read the file into memory. You can set the request body to be a reader that reads from the temp file.

Reply:
changed to a file reader for the body